### ISSUE TITLE
Build fails after recent changes in build.sbt

### ISSUE THREAD

**Issue by frgomes** [scala-native/scala-native-example#7]
The following instructions *should* just work:

    $ find $HOME/.ivy2 -type d -name '*scala-native*' | xargs rm -r -f
    $ sbt clean package nativeLink

Observe that I'm making sure that we clean whatever we have in the local Ivy repository before we start. **This is a critical step!** i.e: always, always! make sure we build from a clean slate!

People trying Scala Native for the first time will be building their first app from a completely clean slate. So we must guarantee we test our build from a complete clean slate.

By the way, it's critical that the documentation is **always in sync** with the build script. Otherwise, people trying Scala Native for the first time will have troubles, will not be able to fix it (obviously, since they are complete newbies on the matter!) and will end up asking things in the gitter room.

In other words, before commiting changes, the entire build must be performed in a clean slate. If things fail (which is the case at the moment), the change cannot be pushed onto the repository otherwise new users (and even regular users!) will be doomed to face troubles.

**Comment by frgomes** [comment]
CC: @densh @andreaTP

**Comment by frgomes** [comment]
The build fails like this below:
```bash
$ find $HOME/.ivy2 -type d -name '*scala-native*' | xargs rm -r -f
$ sbt clean package nativeLink
...
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.scala-native#nativelib_2.11;0.1-SNAPSHOT: not found
[warn]  :: org.scala-native#javalib_2.11;0.1-SNAPSHOT: not found
[warn]  :: org.scala-native#scalalib_2.11;0.1-SNAPSHOT: not found
[warn]  :: org.scala-native#nscplugin_2.11.8;0.1-SNAPSHOT: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn] 
[warn]  Note: Unresolved dependencies path:
[warn]          org.scala-native:nativelib_2.11:0.1-SNAPSHOT ((scala.scalanative.sbtplugin.ScalaNativePluginInternal) ScalaNativePluginInternal.scala#L62)
[warn]            +- default:scala-native-example_2.11:0.1-SNAPSHOT
[warn]          org.scala-native:javalib_2.11:0.1-SNAPSHOT ((scala.scalanative.sbtplugin.ScalaNativePluginInternal) ScalaNativePluginInternal.scala#L62)
[warn]            +- default:scala-native-example_2.11:0.1-SNAPSHOT
[warn]          org.scala-native:scalalib_2.11:0.1-SNAPSHOT ((scala.scalanative.sbtplugin.ScalaNativePluginInternal) ScalaNativePluginInternal.scala#L62)
[warn]            +- default:scala-native-example_2.11:0.1-SNAPSHOT
[warn]          org.scala-native:nscplugin_2.11.8:0.1-SNAPSHOT ((sbt.BuildExtra) Defaults.scala#L1961)
[warn]            +- default:scala-native-example_2.11:0.1-SNAPSHOT
```

I suppose it has something to do with removed lines:
```scala
      compilerPlugin("org.scala-native" %  "tools_2.10" % "0.1-SNAPSHOT"),
      compilerPlugin("org.scala-native" %  "nir_2.10"   % "0.1-SNAPSHOT"),
      compilerPlugin("org.scala-native" %  "util_2.10"  % "0.1-SNAPSHOT"),
```

For your information, the original project in my account builds properly as per its documentation in the README.md. http://github.com/frgomes/scala-native-example

**Comment by frgomes** [comment]
I've reviewed the documentation and *build.sbt* so that users following the old procedure of building Scala Native first and only after that, building the example application... these users will succeed.

**Comment by densh** [comment]
I'm looking at the issue right now, I suspect that https://github.com/scala-native/scala-native/pull/230 should fix this, but we need wait a bit for main repo to update latest snapshot.

**Comment by densh** [comment]
Seems to be fixed now:
```
# find $HOME/.ivy2 -type d -name '*scala-native*' | xargs rm -r -f
# sbt clean nativeLink
[info] Loading project definition from /Users/Denys/.src/native-example/project
[info] Updating {file:/Users/Denys/.src/native-example/project/}native-example-build...
[info] Resolving org.scala-native#sbtplugin;0.1-SNAPSHOT ...
[warn] Unable to reparse org.scala-native#sbtplugin;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:20 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#sbtplugin;0.1-SNAPSHOT
[info] Resolving org.scala-native#tools_2.10;0.1-SNAPSHOT ...
[warn] Unable to reparse org.scala-native#tools_2.10;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:17 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#tools_2.10;0.1-SNAPSHOT
[info] Resolving org.scala-native#nir_2.10;0.1-SNAPSHOT ...
z[warn] Unable to reparse org.scala-native#nir_2.10;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:15 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#nir_2.10;0.1-SNAPSHOT
[info] Resolving org.scala-native#util_2.10;0.1-SNAPSHOT ...
^[[warn] Unable to reparse org.scala-native#util_2.10;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:13 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#util_2.10;0.1-SNAPSHOT
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/sbtplugin_2.10_0.13/0.1-SNAPSHOT/sbtplugin-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#sbtplugin;0.1-SNAPSHOT!sbtplugin.jar (652ms)
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/tools_2.10/0.1-SNAPSHOT/tools_2.10-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#tools_2.10;0.1-SNAPSHOT!tools_2.10.jar (1057ms)
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/nir_2.10/0.1-SNAPSHOT/nir_2.10-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#nir_2.10;0.1-SNAPSHOT!nir_2.10.jar (750ms)
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/util_2.10/0.1-SNAPSHOT/util_2.10-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#util_2.10;0.1-SNAPSHOT!util_2.10.jar (555ms)
[info] Done updating.
[info] Set current project to native-example (in build file:/Users/Denys/.src/native-example/)
[success] Total time: 0 s, completed Jul 7, 2016 3:06:21 PM
[info] Updating {file:/Users/Denys/.src/native-example/}native-example...
[info] Resolving org.scala-native#nativelib_2.11;0.1-SNAPSHOT ...
[warn] Unable to reparse org.scala-native#nativelib_2.11;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:15 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#nativelib_2.11;0.1-SNAPSHOT
[info] Resolving org.scala-native#javalib_2.11;0.1-SNAPSHOT ...
[warn] Unable to reparse org.scala-native#javalib_2.11;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:16 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#javalib_2.11;0.1-SNAPSHOT
[info] Resolving org.scala-native#scalalib_2.11;0.1-SNAPSHOT ...
[warn] Unable to reparse org.scala-native#scalalib_2.11;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:18 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#scalalib_2.11;0.1-SNAPSHOT
[info] Resolving org.scala-native#nscplugin_2.11.8;0.1-SNAPSHOT ...
[warn] Unable to reparse org.scala-native#nscplugin_2.11.8;0.1-SNAPSHOT from sonatype-snapshots, using Thu Jul 07 15:05:14 CEST 2016
[warn] Choosing sonatype-snapshots for org.scala-native#nscplugin_2.11.8;0.1-SNAPSHOT
[info] Resolving jline#jline;2.12.1 ...
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/nativelib_2.11/0.1-SNAPSHOT/nativelib_2.11-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#nativelib_2.11;0.1-SNAPSHOT!nativelib_2.11.jar (1151ms)
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/javalib_2.11/0.1-SNAPSHOT/javalib_2.11-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#javalib_2.11;0.1-SNAPSHOT!javalib_2.11.jar (1088ms)
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/scalalib_2.11/0.1-SNAPSHOT/scalalib_2.11-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#scalalib_2.11;0.1-SNAPSHOT!scalalib_2.11.jar (2500ms)
[info] downloading https://oss.sonatype.org/content/repositories/snapshots/org/scala-native/nscplugin_2.11.8/0.1-SNAPSHOT/nscplugin_2.11.8-0.1-SNAPSHOT.jar ...
[info]  [SUCCESSFUL ] org.scala-native#nscplugin_2.11.8;0.1-SNAPSHOT!nscplugin_2.11.8.jar (625ms)
[info] Done updating.
[info] Compiling 1 Scala source to /Users/Denys/.src/native-example/target/scala-2.11/classes...
warning: overriding the module target triple with x86_64-apple-macosx10.11.0 [-Woverride-module]
1 warning generated.
[success] Total time: 14 s, completed Jul 7, 2016 3:06:36 PM
```
